### DISCUSSION TITLE
Background color for Group draws over child widget

### DISCUSSION THREAD

**Discussion by wyhinton** [fltk-rs/fltk-rs#695] [Q&A] [answered]
I'm trying to give my ```Group``` widget a background color.  I thought calling ```draw_rect_fill``` and _then_  calling ```draw_children()``` in my group's ```draw``` call would display the child widgets _on top of_ the background fill color. This doesn't seem to be the case:

```lang-rust
//fltk = "1.0.14"
use fltk::{app, prelude::*, frame::Frame, window::Window, enums::*, group::Group, draw::*};

fn main() {
    let app = app::App::default();
    let mut wind = Window::default().with_size(400, 300).with_label("Group Background Color");
    wind.make_resizable(true);
    let mut group = Group::new(0,0,400,300, None);

    //frame we want to see
    let mut frame = Frame::new(0,0,100,100, "");
    frame.set_color(Color::Magenta);
    frame.set_frame(FrameType::FlatBox);
    group.end();
    group.draw(move |widg| {
        //background fill
        draw_rect_fill(
            widg.x(),
            widg.y(),
            widg.width(),
            widg.height(),
            Color::Red,
        );
        //draw the children after drawing the background fill color
        widg.draw_children();
    });
    wind.end();
    wind.show();
    app.run().unwrap();
}
```
Gives:
![ActualOutput](https://user-images.githubusercontent.com/38958118/119585188-2ea25000-bd98-11eb-9e97-38bf833f9770.PNG)
 As you can see, our magenta frame widget is no where to be found.

Desired output, for reference:
![DesiredOutput](https://user-images.githubusercontent.com/38958118/119585212-39f57b80-bd98-11eb-8c61-3b91e224ba53.PNG)

Anybody have an approach for this, or understand why this doesn't work as expected?

**Comment by santokalayil** [comment]
Same issue I also found. I think there we need to use pack.

**Reply by wyhinton** [reply]
Unfortunately this means having to get around the auto-resizing that ```Pack``` applies. In the past, I've nested a colored ```Frame``` inside of a ```Group```, inside of a ```Pack```. This is incredibly verbose for what is a pretty simple task. I'm just stumped as to why the children are not on top of the fill, are the draw functions not executed sequentially?

**Reply by santokalayil** [reply]
I know not.

**Comment by MoAlyousef** [comment, marked as answer]
Not sure, but the code seems to work correctly on my machine (tested on Windows 10 x64 and WSL Ubuntu 20.04).

**Reply by wyhinton** [reply]
Hmm yes it seems to be working now for me as well, not sure what happened. I'll add this to my [examples repo](https://github.com/wyhinton/FLTK-RS-Examples).
